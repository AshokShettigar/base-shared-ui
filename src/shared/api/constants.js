export const partnerId = '0';

// https://continuum.atlassian.net/wiki/spaces/C2E/pages/153880042/REST+API+Design+Guide
export const HTTP_STATUSES = {
    OK: 200,
    CREATED: 201,
    NO_CONTENT: 204,
    BAD_REQUEST: 400,
    UNAUTHORIZED: 401,
    FORBIDDEN: 403,
    NOT_FOUND: 404,
    METHOD_NOT_ALLOWED: 405,
    CONFLICT: 409,
    UNSUPPORTED_MEDIA_TYPE: 415,
    TOO_MANY_REQUESTS: 429,
    CLIENT_CLOSED_REQUEST: 499,
    INTERNAL_SERVER_ERROR: 500,
    SERVICE_UNAVAILABLE: 503
};

export const URLS = {
    INTEGRATION: 'https://integration.control.v2.itsupport247.net',
    LOG_OUT: 'https://control.dtitsupport247.net/logout.asp'
};
