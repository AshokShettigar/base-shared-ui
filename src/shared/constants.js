export const SIZES = {
    MINI: 'mini',
    TINY: 'tiny',
    SMALL: 'small',
    MEDIUM: 'medium',
    LARGE: 'large',
    BIG: 'big',
    HUGE: 'huge',
    MASSIVE: 'massive',
    BANNER: 'banner'
};

export const KINDS = {
    PRIMARY: 'primary',
    SECONDARY: 'secondary'
};
