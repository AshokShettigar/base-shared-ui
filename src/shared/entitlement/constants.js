export default {
    TASKING_BASIC: 'TASKING_BASIC',
    SEQUENCE_BASIC: 'SEQUENCE_BASIC',
    TASKING_DYNAMIC_GROUPS: 'TASKING_DYNAMIC_GROUPS',
    TASKING_TASKS_HOME_PAGE: 'TASKING_TASKS_HOME_PAGE',
    PATCHING_DESKTOPS: 'Desktops Patching UI',
    PATCHING_APP_FRESH_INSTALL: 'AppFreshInstall',
    PATCHING_DEVICE_LEVEL_ASSIGNMENT: 'PatchingDeviceLevelAssignment'
};
