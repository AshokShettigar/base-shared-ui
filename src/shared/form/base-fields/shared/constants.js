export const DAYS_OF_WEEK = [{
    shortName: 'Sun',
    name: 'Sunday',
    number: 0
}, {
    shortName: 'Mon',
    name: 'Monday',
    number: 1
}, {
    shortName: 'Tue',
    name: 'Tuesday',
    number: 2
}, {
    shortName: 'Wed',
    name: 'Wednesday',
    number: 3
}, {
    shortName: 'Thu',
    name: 'Thursday',
    number: 4
}, {
    shortName: 'Fri',
    name: 'Friday',
    number: 5
}, {
    shortName: 'Sat',
    name: 'Saturday',
    number: 6
}];
