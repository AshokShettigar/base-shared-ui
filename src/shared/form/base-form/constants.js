export const SELECTED_FIELD = 'SELECTED_FIELD';
