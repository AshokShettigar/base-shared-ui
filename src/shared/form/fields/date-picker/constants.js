export const defaultDateFormat = 'DD/MM/YYYY';
export const defaultTimeFormat = 'hh:mm A';

export const defaultDatePlaceholder = defaultDateFormat;
export const defaultTimePlaceholder = 'HH:MM';

export const VARIATION = {
    TIME: 'time',
    DATE: 'date',
    DATE_TIME: 'dateTime'
};
