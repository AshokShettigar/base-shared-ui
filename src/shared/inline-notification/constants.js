export const NOTIFICATION_ICON_TYPES = {
    info: 'info',
    warning: 'alert',
    error: 'remove',
    success: 'success',
    message: 'message'
};

export const ROOT_CLASS = 'inline-notification';
