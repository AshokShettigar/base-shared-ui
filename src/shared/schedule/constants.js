import { map } from 'lodash';
import { DAYS_OF_WEEK } from '../form';

export { SELECTED_FIELD } from '../form';

export const UNSPECIFIC_INDICATORS = [{
    label: 'First',
    value: 'first'
}, {
    label: 'Second',
    value: 'second'
}, {
    label: 'Third',
    value: 'third'
}, {
    label: 'Fourth',
    value: 'fourth'
}, {
    label: 'Fifth',
    value: 'fifth'
}, {
    label: 'Last',
    value: 'last'
}];

export const unspecificDaysOptions = map(DAYS_OF_WEEK, date => ({
    label: date.name,
    value: date.number
}));
