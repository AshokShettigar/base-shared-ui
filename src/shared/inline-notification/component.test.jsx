import { shallow } from 'enzyme';
import InlineNotification from './component';

import { ROOT_CLASS } from './constants';


const defaultProps = {
    isOpen: true,
    type: 'message'
};

describe('Inline Notification', () => {
    it('default renders', () => {
        const wrapper = shallow(
            <InlineNotification {...defaultProps}>
                Inline Notification Message
            </InlineNotification>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with type warning', () => {
        const wrapper = shallow(
            <InlineNotification {...defaultProps} type="warning">
                Inline Notification Warning Message
            </InlineNotification>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with type error', () => {
        const wrapper = shallow(
            <InlineNotification {...defaultProps} type="error">
                Inline Notification Error Message
            </InlineNotification>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders with type info', () => {
        const wrapper = shallow(
            <InlineNotification {...defaultProps} type="info">
                Inline Notification Info Message
            </InlineNotification>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with type success', () => {
        const wrapper = shallow(
            <InlineNotification {...defaultProps} type="success">
                Inline Notification Success Message
            </InlineNotification>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders closed', () => {
        const wrapper = shallow(
            <InlineNotification {...defaultProps} isOpen={false}>
                Inline Notification Message
            </InlineNotification>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with custom className props', () => {
        const customProps = {
            buttonCloseClassName: 'button-close-class-name',
            className: 'class-name',
            contentClassName: 'content-class-name',
            iconTypeClassName: 'icon-type-class-name',
            iconCloseClassName: 'icon-close-class-name'
        };

        const wrapper = shallow(
            <InlineNotification {...defaultProps} {...customProps}>
                Inline Notification Message
            </InlineNotification>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders with handler', () => {
        const props = {
            onClose: jest.fn()
        };

        const wrapper = shallow(
            <InlineNotification {...defaultProps} {...props}>
                Inline Notification Message
            </InlineNotification>
        );

        wrapper.find(`.${ROOT_CLASS}__button-close`).simulate('click');

        expect(props.onClose).toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
    });
});

