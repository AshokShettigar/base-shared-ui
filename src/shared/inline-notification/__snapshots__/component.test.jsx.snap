// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline Notification default renders 1`] = `
<div
  className="inline-notification inline-notification--message"
  data-component="InlineNotification"
>
  <Icon
    className="inline-notification__icon inline-notification__icon--message"
    icon="message"
    size="extra-small"
  />
  <div
    className="inline-notification__content"
  >
    Inline Notification Message
  </div>
</div>
`;

exports[`Inline Notification renders closed 1`] = `null`;

exports[`Inline Notification renders with custom className props 1`] = `
<div
  className="inline-notification inline-notification--message class-name"
  data-component="InlineNotification"
>
  <Icon
    className="inline-notification__icon inline-notification__icon--message icon-type-class-name"
    icon="message"
    size="extra-small"
  />
  <div
    className="inline-notification__content content-class-name"
  >
    Inline Notification Message
  </div>
</div>
`;

exports[`Inline Notification renders with handler 1`] = `
<div
  className="inline-notification inline-notification--message"
  data-component="InlineNotification"
>
  <Icon
    className="inline-notification__icon inline-notification__icon--message"
    icon="message"
    size="extra-small"
  />
  <div
    className="inline-notification__content"
  >
    Inline Notification Message
  </div>
  <Button
    className="cn-link inline-notification__button-close"
    isDark={false}
    isDisabled={false}
    onClick={[Function]}
    size={1}
    type="button"
    variation={0}
  >
    <Icon
      className="inline-notification__icon inline-notification__icon--close"
      icon="close"
      size="extra-small"
    />
  </Button>
</div>
`;

exports[`Inline Notification renders with type error 1`] = `
<div
  className="inline-notification inline-notification--error"
  data-component="InlineNotification"
>
  <Icon
    className="inline-notification__icon inline-notification__icon--error"
    icon="remove"
    size="extra-small"
  />
  <div
    className="inline-notification__content"
  >
    Inline Notification Error Message
  </div>
</div>
`;

exports[`Inline Notification renders with type info 1`] = `
<div
  className="inline-notification inline-notification--info"
  data-component="InlineNotification"
>
  <Icon
    className="inline-notification__icon inline-notification__icon--info"
    icon="info"
    size="extra-small"
  />
  <div
    className="inline-notification__content"
  >
    Inline Notification Info Message
  </div>
</div>
`;

exports[`Inline Notification renders with type success 1`] = `
<div
  className="inline-notification inline-notification--success"
  data-component="InlineNotification"
>
  <Icon
    className="inline-notification__icon inline-notification__icon--success"
    icon="success"
    size="extra-small"
  />
  <div
    className="inline-notification__content"
  >
    Inline Notification Success Message
  </div>
</div>
`;

exports[`Inline Notification renders with type warning 1`] = `
<div
  className="inline-notification inline-notification--warning"
  data-component="InlineNotification"
>
  <Icon
    className="inline-notification__icon inline-notification__icon--warning"
    icon="alert"
    size="extra-small"
  />
  <div
    className="inline-notification__content"
  >
    Inline Notification Warning Message
  </div>
</div>
`;
